### DISCUSSION TITLE
Is there a way to override the @TimeStamp that is set by the Server?

### DISCUSSION THREAD

**Discussion by giblets** [datalust/seq-tickets#1893] [Help] [answered]
I'm trying to take a Log table in SQL Server and feed the contents into Seq.  I want to do this so I can demo how awesome it is, and how we could use it to visualize the events we are currently only logging to a table.  

To do this I want to show them what a real day, week, or month's worth of data would be like to look at, filter and work with.  I have a Powershell script to loop through the table and log the contents to a Seq Server I have running on Docker.  I want the TimeStamps in the Log table to be the @TimeStamps for the Seq Server.

Is there a way we can do this?  

From the Documentation it looks like maybe the SeqCLI can do this, but I'm running this from a Docker Container and for security reasons it's difficult to get software installed on my local machine.  

Thanks!

**Comment by liammclennan** [comment, marked as answer]
Hi @giblets 

If your powershell script is using Serilog to send the log events to Seq then you can use the `LogEvent` constructor that includes a timestamp and the `ILogger.Write(LogEvent)` function to send the event. If you can send the events in order it will be more efficient on the Seq side.

**Comment by giblets** [comment]
Thank you Liam!  Forget everything bad I ever said about you!  

This opens up Seq to doing historical log ingestion!  Perfect for my demo, and very useful for beyond that!  ...I'm guessing it might risk thrashing the Indexing though?

Thank you!

**Reply by liammclennan** [reply]
🤣 

Performance will be reduced until the log has been re-indexed. Then it will be fine.
